### DISCUSSION TITLE
How to install on Ubuntu 24.04

### DISCUSSION THREAD

**Discussion by NsinghP** [jziolkowski/tdm#265] [Q&A] [answered]
Hello, I am already using TDM on windows 11 and it works great with mqtt on my Home Assistant Raspberry Pi device. I have recently started experimenting with Ubuntu/Linux. I was not able to find anything on the net that gives me step by step instructions on how to install TDM on Ubuntu 24.04 LTS. So far it is my understanding that if Python is installed on the Ubuntu system then all I need to do is use the command `pip install tdmgr` and after the installation is done all I need to launch TDM is by using the command `tdmgr.py `in ubuntu console? All helpful hints to get TDM working on Ubuntu will be appreciated. 

Thanks. 
Navi

**Comment by jziolkowski** [comment]
That sounds about right. If you don't want to add TDM's requirements to your system install, consider creating a virtualenv for TDM.

**Reply by NsinghP** [reply]
Thanks for your reply. After installing TDM on Ubunuto 24.04 I was able to launch it with `tdmgr.py` command. However, when I try to connect with MQTT server, TDM crashes. I am assuming it is just not compatible with Ubuntu 24.04  or maybe I did something wrong when installing. Any pointers on resolving the crash will be appreciated. Below are the details of this crash:

ubuntu:~$ `tdmgr.py`
```
home/n/.local/bin/tdmgr.py:45: DeprecationWarning: sipPyTypeDict() is deprecated, the extension module should use sipPyTypeDictRef() instead
  class MainWindow(QMainWindow):
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
```
At this point the TDM WEB GUI is visible. After I enter my MQTT server info and hit the connect button the TDM WEBGui vanishes (I am assuming it crashes). and leaves following messages on the terminal:

```
Traceback (most recent call last):
  File "/home/n/.local/share/pipx/venvs/tdmgr/lib/python3.12/site-packages/Util/models.py", line 281, in paint
    p.drawPixmap(option.rect.x() + 2, option.rect.y() + px_y, px.scaled(24, 24))
TypeError: arguments did not match any overloaded call:
  drawPixmap(self, QRectF, QPixmap, QRectF): argument 1 has unexpected type 'int'
  drawPixmap(self, QRect, QPixmap, QRect): argument 1 has unexpected type 'int'
  drawPixmap(self, Union[QPointF, QPoint], QPixmap): argument 1 has unexpected type 'int'
  drawPixmap(self, QPoint, QPixmap): argument 1 has unexpected type 'int'
  drawPixmap(self, QRect, QPixmap): argument 1 has unexpected type 'int'
  drawPixmap(self, int, int, QPixmap): argument 2 has unexpected type 'float'
  drawPixmap(self, int, int, int, int, QPixmap): argument 2 has unexpected type 'float'
  drawPixmap(self, int, int, int, int, QPixmap, int, int, int, int): argument 2 has unexpected type 'float'
  drawPixmap(self, int, int, QPixmap, int, int, int, int): argument 2 has unexpected type 'float'
  drawPixmap(self, Union[QPointF, QPoint], QPixmap, QRectF): argument 1 has unexpected type 'int'
  drawPixmap(self, QPoint, QPixmap, QRect): argument 1 has unexpected type 'int'
Aborted (core dumped)

```

Thanks for your time.
--Navi

**Reply by jziolkowski** [reply, marked as answer]
uninstall TDM and get the deelopment version from github. I keep forgetting to make a release with the bugfix for that

**Reply by NsinghP** [reply]
> uninstall TDM and get the deelopment version from github. I keep forgetting to make a release with the bugfix for that

Ok thanks. I will try. First I have to figure out how to uninstall :)  and then how to get development version. By the way I was reading up on discussion in Issues and saw discussion #261 on debian package. I have downloaded the .deb file, I may try the debian file after I figure out how to uninstall tdmgr. I will post my results here. Thanks.

**Reply by NsinghP** [reply]
Installed the downloaded .deb file using `sudo apt install `command. Now TDM is working on my Ubuntu 24.04 system.  Here is the terminal readout when I executed the intall command:

```
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'tasmota-device-manager' instead of '/home/n/Downloads/tasmota-device-manager_0.2.13-1_all.deb'
The following additional packages will be installed:
  python3-paho-mqtt
The following NEW packages will be installed:
  python3-paho-mqtt tasmota-device-manager
0 upgraded, 2 newly installed, 0 to remove and 22 not upgraded.
Need to get 52.9 kB/141 kB of archives.
After this operation, 695 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 /home/n/Downloads/tasmota-device-manager_0.2.13-1_all.deb tasmota-device-manager all 0.2.13-1 [88.5 kB]
Get:2 http://in.archive.ubuntu.com/ubuntu noble/universe amd64 python3-paho-mqtt all 1.6.1-1 [52.9 kB]
Fetched 52.9 kB in 1s (53.0 kB/s)            
Selecting previously unselected package python3-paho-mqtt.
(Reading database ... 218817 files and directories currently installed.)
Preparing to unpack .../python3-paho-mqtt_1.6.1-1_all.deb ...
Unpacking python3-paho-mqtt (1.6.1-1) ...
Selecting previously unselected package tasmota-device-manager.
Preparing to unpack .../tasmota-device-manager_0.2.13-1_all.deb ...
Unpacking tasmota-device-manager (0.2.13-1) ...
Setting up python3-paho-mqtt (1.6.1-1) ...
Setting up tasmota-device-manager (0.2.13-1) ...
Processing triggers for desktop-file-utils (0.27-2build1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for gnome-menus (3.36.0-1.1ubuntu3) ...
Processing triggers for man-db (2.12.0-4build2) ...
N: Download is performed unsandboxed as root as file '/home/n/Downloads/tasmota-device-manager_0.2.13-1_all.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)
```
I hope this method of installation does not cause any problem later. Any hints/pointers will be appreciate. Thanks for your  time.

**Reply by jziolkowski** [reply]
This looks as expected. The last error can be mitigated by installing from /tmp, to which user `_apt` has access. Other than that looks fine.
